drivers/include/sx126x.h

```diff
@@ -97,6 +99,23 @@ void sx126x_setup(sx126x_t *dev, const sx126x_params_t *params, uint8_t index);
  */
 int sx126x_init(sx126x_t *dev);
 
+/**
+ * @brief   Converts symbol value to time in milliseconds.
+ *
+ * @param[in] dev                      Device descriptor of the driver
+ * @param[in] symbols                  Symbols
+ *
+ * @return Time for symbol(s) in milliseconds
+ */
+static inline int sx126x_symbol_to_msec(sx126x_t *dev, uint16_t symbols)
+{
+    assert(dev && (dev->mod_params.bw <= SX126X_LORA_BW_500) && \
+           (dev->mod_params.bw >= SX126X_LORA_BW_125));
+
+    /* Refer section 4.1.1.7  Time on air in SX1276/77/78/79 datasheet */
+    return (symbols * (1 << (dev->mod_params.sf + 7 - dev->mod_params.bw)) / 1000);
```

Comment:
This formula seems wrong:

![image](https://user-images.githubusercontent.com/23060007/124892192-f23e5280-dfd9-11eb-8122-29db1c12e604.png)

Reply:
It's a simplified version, just put in the numbers and the results are the same.

Reply:
How? dev->mod_params.bw is not the bandwidth in KHz  but an enum and those values do not easily translate to Khz do they?

```
 * @brief SX126X LoRa bandwidth enumeration definition
 */
typedef enum sx126x_lora_bw_e
{
    SX126X_LORA_BW_500 = 6,
    SX126X_LORA_BW_250 = 5,
    SX126X_LORA_BW_125 = 4,
    SX126X_LORA_BW_062 = 3,
    SX126X_LORA_BW_041 = 10,
    SX126X_LORA_BW_031 = 2,
    SX126X_LORA_BW_020 = 9,
    SX126X_LORA_BW_015 = 1,
    SX126X_LORA_BW_010 = 8,
    SX126X_LORA_BW_007 = 0,
} sx126x_lora_bw_t;
```

And for the `sf`:

```
/**
 * @brief SX126X LoRa spreading factor enumeration definition
 */
typedef enum sx126x_lora_sf_e
{
    SX126X_LORA_SF5  = 0x05,
    SX126X_LORA_SF6  = 0x06,
    SX126X_LORA_SF7  = 0x07,
    SX126X_LORA_SF8  = 0x08,
    SX126X_LORA_SF9  = 0x09,
    SX126X_LORA_SF10 = 0x0A,
    SX126X_LORA_SF11 = 0x0B,
    SX126X_LORA_SF12 = 0x0C,
} sx126x_lora_sf_t;
```

I'm only asking because @jia200x pointed out that timings are wrong... so I'm trying to understand why, but if you are sure its correct then the issue is somwhere else I guess

Reply:
`1 / BW[MHz] = (1 / 2^(i - 7)`, with i the BW index (4: 125, 5: 250, 6: 500), as defined by the driver.

Therefore  `Ts = 2^(SF) * (1 / 2^(i - 7))  = 2^(SF + 7 - i) `, for SF={7,8,9,10,11,12} and i={4,5,6}.

I tested the formula with this python script:
```python
SF = [7,8,9,10,11,12]
BW = [4,5,6]
for b in BW:
    for s in SF:
        Ts = (1 << (s + 7 - b))/1000
        print("SF:{},BW:{}->Ts = {}".format(s,b,Ts))
```

and throws:
```
SF:7,BW:4->Ts = 1.024
SF:8,BW:4->Ts = 2.048
SF:9,BW:4->Ts = 4.096
SF:10,BW:4->Ts = 8.192
SF:11,BW:4->Ts = 16.384
SF:12,BW:4->Ts = 32.768
SF:7,BW:5->Ts = 0.512
SF:8,BW:5->Ts = 1.024
SF:9,BW:5->Ts = 2.048
SF:10,BW:5->Ts = 4.096
SF:11,BW:5->Ts = 8.192
SF:12,BW:5->Ts = 16.384
SF:7,BW:6->Ts = 0.256
SF:8,BW:6->Ts = 0.512
SF:9,BW:6->Ts = 1.024
SF:10,BW:6->Ts = 2.048
SF:11,BW:6->Ts = 4.096
SF:12,BW:6->Ts = 8.192
```

We should check that the index are right though (e.g SF=7 maps to `dev->mod_params.sf == 7`)

Reply:
@akshaim could you add print messages for the indexes and the output value and send the results around?

Reply:
It seems its an issue with the patch, the SF is not updated from SF7

Reply:
HAhahhaa layer 8, its set not get...

Reply:
uuuuf that was my fault :O